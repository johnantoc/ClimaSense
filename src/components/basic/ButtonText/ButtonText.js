/**
 * Button Component.
 */

import React from "react";
import { Text, View, TouchableOpacity } from "react-native";

import ButtonTextStyles from "./ButtonText.style";

const ButtonText = ({ title, onPressHandler, textStyles }) => {
  return (
    <TouchableOpacity
      activeOpacity={0.8}
      style={ButtonTextStyles.buttonView}
      onPress={onPressHandler}
    >
      <Text style={[ButtonTextStyles.button, textStyles]}>{title}</Text>
    </TouchableOpacity>
  );
};

export default ButtonText;
