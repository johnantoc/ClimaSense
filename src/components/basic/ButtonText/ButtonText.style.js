import { StyleSheet } from "react-native";

const ButtonTextStyles = StyleSheet.create({
  buttonView: {
    flexDirection: "row",
    justifyContent: "center",
  },
  button: {
    fontFamily: "roboto-regular",
  },
});

export default ButtonTextStyles;
