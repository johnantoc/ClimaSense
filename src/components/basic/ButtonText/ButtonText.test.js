import React from 'react';
import ButtonText from './ButtonText';

import renderer from 'react-test-renderer';

test('ButtonText renders correctly', () => {
  const tree = renderer
    .create(<ButtonText title="Details" onPressHandler={() => null} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
