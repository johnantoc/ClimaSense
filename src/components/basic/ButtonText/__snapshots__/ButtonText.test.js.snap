// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonText renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="btnGoToDetails"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Details
      </Text>
    </View>
  </View>
</View>
`;
